src/core/thread/mle.cpp

```diff
@@ -627,6 +634,11 @@ otError Mle::SetStateChild(uint16_t aRloc16)
         netif.GetAnnounceBeginServer().SendAnnounce(1 << mPreviousChannel);
     }
 
+#if OPENTHREAD_CONFIG_INFORM_PREVIOUS_PARENT_ON_REATTACH
+    InformPreviousParent();
```

Comment:
Should this be moved to `HandleChildIdResponse()`?  `SetStateChild()` is called from a number of places, but only `HandleChildIdResponse()` will occur when attaching to a different parent.

Reply:
I think this is safer in the sense that no matter how we get re-attached we go through this part of the code.  
We already have  checks in `InformPreviousParent()` to make sure we don't send to same parent or if the previous parent is invalid.